File: android/app/src/main/java/com/nanoai/ui/presentation/settings/SettingsScreen.kt
package com.nanoai.ui.presentation.settings

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.nanoai.ui.theme.*

/**
 * SettingsScreen (Sección 7 Specification Compliant — 7 Secciones)
 */
@Composable
fun SettingsScreen(
    viewModel: SettingsViewModel,
    modifier: Modifier = Modifier
) {
    val state by viewModel.uiState.collectAsState()

    SettingsContent(
        state = state,
        onAction = viewModel::onAction,
        modifier = modifier
    )
}

@Composable
fun SettingsContent(
    state: SettingsUiState,
    onAction: (SettingsUiAction) -> Unit,
    modifier: Modifier = Modifier
) {
    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .padding(horizontal = NanoSpacingTokens.lg),
        verticalArrangement = Arrangement.spacedBy(NanoSpacingTokens.lg)
    ) {
        // Section 1: General
        item {
            SettingsSectionHeader(title = "1. GENERAL", icon = Icons.Default.Tune)
            SettingsSelectorRow(
                title = "Tema de la aplicación",
                subtitle = "Claro, oscuro o automático según el sistema",
                options = listOf("Sistema", "Oscuro", "Claro"),
                selected = state.themeMode,
                onOptionSelected = { onAction(SettingsUiAction.UpdateThemeMode(it)) }
            )
        }

        // Section 2: Memoria & RAM
        item {
            SettingsSectionHeader(title = "2. GESTIÓN DE MEMORIA (RAM)", icon = Icons.Default.Memory)
            SettingsSwitchRow(
                title = "madvise(DONTNEED) Layer Streaming",
                subtitle = "Descarga quirúrgica por capa (VMA < 1GB)",
                checked = state.enableMadviseLayerStreaming,
                onCheckedChange = { onAction(SettingsUiAction.ToggleMadvise(it)) }
            )
            SettingsSwitchRow(
                title = "OOM Guard System",
                subtitle = "Activa modo supervivencia antes del crash",
                checked = state.enableOomGuard,
                onCheckedChange = { onAction(SettingsUiAction.ToggleOomGuard(it)) }
            )
        }

        // Section 3: Control Térmico
        item {
            SettingsSectionHeader(title = "3. CONTROL TÉRMICO DE SOC", icon = Icons.Default.Thermostat)
            SettingsSliderRow(
                title = "Límite de Temperatura Max (°C)",
                value = state.maxThermalLimitC,
                range = 35f..50f,
                steps = 15,
                unitLabel = "°C",
                onValueChange = { onAction(SettingsUiAction.UpdateThermalLimit(it)) }
            )
        }

        // Section 4: Battery Guardian
        item {
            SettingsSectionHeader(title = "4. GUARDIÁN DE BATERÍA", icon = Icons.Default.BatteryStd)
            Row(
                horizontalArrangement = Arrangement.spacedBy(NanoSpacingTokens.sm),
                modifier = Modifier.fillMaxWidth()
            ) {
                listOf("Eco", "Balanced", "Survival").forEach { mode ->
                    FilterChip(
                        selected = state.batteryMode == mode,
                        onClick = { onAction(SettingsUiAction.UpdateBatteryMode(mode)) },
                        label = { Text(mode) },
                        modifier = Modifier.weight(1f)
                    )
                }
            }
        }

        // Section 5: Inferencia LLM
        item {
            SettingsSectionHeader(title = "5. PARÁMETROS DE INFERENCIA LLM", icon = Icons.Default.Psychology)
            SettingsSliderRow(
                title = "Temperatura (Creatividad)",
                value = state.temperature,
                range = 0.1f..1.5f,
                steps = 14,
                unitLabel = "",
                onValueChange = { onAction(SettingsUiAction.UpdateTemperature(it)) }
            )
            SettingsSliderRow(
                title = "Top-P Sampling",
                value = state.topP,
                range = 0.1f..1.0f,
                steps = 9,
                unitLabel = "",
                onValueChange = { onAction(SettingsUiAction.UpdateTopP(it)) }
            )
        }

        // Section 6: Privacidad & Enrutamiento
        item {
            SettingsSectionHeader(title = "6. PRIVACIDAD & ENTROPY ROUTING", icon = Icons.Default.Security)
            SettingsSwitchRow(
                title = "Filtro PII de Datos Sensibles",
                subtitle = "Anonimiza emails y credenciales localmente",
                checked = state.enablePiiAnonymizer,
                onCheckedChange = { onAction(SettingsUiAction.TogglePiiAnonymizer(it)) }
            )
        }

        // Section 7: Avanzado & GPU
        item {
            SettingsSectionHeader(title = "7. CONFIGURACIÓN AVANZADA & GPU", icon = Icons.Default.DeveloperMode)
            SettingsSliderRow(
                title = "Capas GPU Offload",
                value = state.gpuLayers.toFloat(),
                range = 0f..32f,
                steps = 32,
                unitLabel = "capas",
                onValueChange = { onAction(SettingsUiAction.UpdateGpuLayers(it.toInt())) }
            )
        }

        item {
            OutlinedButton(
                onClick = { onAction(SettingsUiAction.ResetToDefaults) },
                modifier = Modifier.fillMaxWidth(),
                colors = ButtonDefaults.outlinedButtonColors(contentColor = MaterialTheme.colorScheme.error)
            ) {
                Icon(imageVector = Icons.Default.Restore, contentDescription = null)
                Spacer(modifier = Modifier.width(NanoSpacingTokens.sm))
                Text("Restablecer Ajustes de Fábrica")
            }
        }
    }
}

@Composable
fun SettingsSectionHeader(title: String, icon: ImageVector) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(NanoSpacingTokens.sm),
        modifier = Modifier.padding(vertical = NanoSpacingTokens.xs)
    ) {
        Icon(imageVector = icon, contentDescription = null, tint = MaterialTheme.colorScheme.primary, modifier = Modifier.size(20.dp))
        Text(text = title, style = NanoTypeTokens.subtitle, color = MaterialTheme.colorScheme.onSurface)
    }
}

@Composable
fun SettingsSwitchRow(
    title: String,
    subtitle: String,
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit
) {
    Card(
        shape = NanoShapeTokens.medium,
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        modifier = Modifier.fillMaxWidth()
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(NanoSpacingTokens.md),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(text = title, style = NanoTypeTokens.body, color = MaterialTheme.colorScheme.onSurface)
                Text(text = subtitle, style = NanoTypeTokens.caption, color = MaterialTheme.colorScheme.onSurfaceVariant)
            }
            Switch(
                checked = checked,
                onCheckedChange = onCheckedChange,
                colors = SwitchDefaults.colors(checkedThumbColor = MaterialTheme.colorScheme.primary)
            )
        }
    }
}

@Composable
fun SettingsSliderRow(
    title: String,
    value: Float,
    range: ClosedFloatingPointRange<Float>,
    steps: Int,
    unitLabel: String,
    onValueChange: (Float) -> Unit
) {
    Card(
        shape = NanoShapeTokens.medium,
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(NanoSpacingTokens.md)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(text = title, style = NanoTypeTokens.body, color = MaterialTheme.colorScheme.onSurface)
                Text(
                    text = String.format("%.1f %s", value, unitLabel),
                    style = NanoTypeTokens.subtitle,
                    color = MaterialTheme.colorScheme.primary
                )
            }
            Slider(
                value = value,
                onValueChange = onValueChange,
                valueRange = range,
                steps = steps,
                colors = SliderDefaults.colors(
                    thumbColor = MaterialTheme.colorScheme.primary,
                    activeTrackColor = MaterialTheme.colorScheme.primary
                )
            )
        }
    }
}

@Composable
fun SettingsSelectorRow(
    title: String,
    subtitle: String,
    options: List<String>,
    selected: String,
    onOptionSelected: (String) -> Unit
) {
    Card(
        shape = NanoShapeTokens.medium,
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        modifier = Modifier.fillMaxWidth()
    ) {
        Column(
            modifier = Modifier.fillMaxWidth().padding(NanoSpacingTokens.md)
        ) {
            Text(text = title, style = NanoTypeTokens.body, color = MaterialTheme.colorScheme.onSurface)
            Text(text = subtitle, style = NanoTypeTokens.caption, color = MaterialTheme.colorScheme.onSurfaceVariant)
            Spacer(Modifier.height(NanoSpacingTokens.sm))
            Row(horizontalArrangement = Arrangement.spacedBy(NanoSpacingTokens.sm)) {
                options.forEach { option ->
                    FilterChip(
                        selected = selected == option,
                        onClick = { onOptionSelected(option) },
                        label = { Text(option, style = NanoTypeTokens.caption) },
                        colors = FilterChipDefaults.filterChipColors(
                            selectedContainerColor = MaterialTheme.colorScheme.primaryContainer,
                            selectedLabelColor = MaterialTheme.colorScheme.onPrimaryContainer
                        ),
                        modifier = Modifier.weight(1f)
                    )
                }
            }
        }
    }
}

File: flutter_app/lib/features/settings/settings_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/theme/design_tokens.dart';
import '../../core/providers/app_providers.dart';

class SettingsScreen extends ConsumerStatefulWidget {
  const SettingsScreen({super.key});
  @override ConsumerState<SettingsScreen> createState() => _SettingsScreenState();
}

class _SettingsScreenState extends ConsumerState<SettingsScreen> {
  // La carga persistida se realiza en main.dart (NanoPlatformApp.initState)
  // para que el tema aplica desde el arranque. Nada que hacer aquí.

  @override
  Widget build(BuildContext context) {
    final state = ref.watch(settingsProvider);
    final notifier = ref.read(settingsProvider.notifier);
    final colors = NanoThemeExtension.of(context).colors;
    final shadow = NanoShadows.card(colors);

    void onTheme(String m) => notifier.setThemeMode(m);

    Widget card(Widget child) => Container(
      width: double.infinity, padding: const EdgeInsets.all(NanoSpacing.lg),
      decoration: BoxDecoration(color: colors.surface, borderRadius: NanoShapes.large, border: Border.all(color: colors.outlineVariant.withValues(alpha: 0.4)), boxShadow: shadow), child: child);

    return ListView(padding: const EdgeInsets.all(NanoSpacing.lg), children: [
        _h('General', Icons.tune, colors),
        card(Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
          Row(children: [Icon(Icons.brightness_6, size: 18, color: colors.primary), const SizedBox(width: NanoSpacing.sm), Text('Tema', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500, color: colors.onSurface))]),
          const SizedBox(height: NanoSpacing.md),
          Row(children: ['Sistema', 'Oscuro', 'Claro'].map((o) => Expanded(child: Padding(padding: const EdgeInsets.only(right: 8), child: ChoiceChip(label: Text(o, style: const TextStyle(fontSize: 11)), selected: state.themeMode == o, onSelected: (_) => onTheme(o), selectedColor: colors.primaryContainer, side: BorderSide.none, shape: RoundedRectangleBorder(borderRadius: NanoShapes.full))))).toList()),
        ])),
        const SizedBox(height: NanoSpacing.md),
        _h('Memoria RAM', Icons.memory, colors),
        card(Column(children: [_sw('madvise Layer Streaming', 'Descarga por capa (VMA < 1GB)', state.madvise, notifier.toggleMadvise, colors), const Divider(height: 1), _sw('OOM Guard System', 'Modo supervivencia pre-crash', state.oomGuard, notifier.toggleOom, colors)])),
        const SizedBox(height: NanoSpacing.md),
        _h('Control Térmico', Icons.thermostat, colors),
        card(_sl('Límite Temperatura Max', state.thermalLimit, 35, 50, '°C', notifier.setThermalLimit, colors)),
        const SizedBox(height: NanoSpacing.md),
        _h('Inferencia LLM', Icons.psychology, colors),
        card(Column(children: [_sl('Temperatura (Creatividad)', state.temperature, 0.1, 1.5, '', notifier.setTemperature, colors), _sl('Top-P Sampling', state.topP, 0.1, 1.0, '', notifier.setTopP, colors)])),
        const SizedBox(height: NanoSpacing.md),
        _h('Batería', Icons.battery_std, colors),
        card(Row(children: ['Eco', 'Balanced', 'Survival'].map((o) => Expanded(child: Padding(padding: const EdgeInsets.only(right: 8), child: ChoiceChip(label: Text(o, style: const TextStyle(fontSize: 11)), selected: state.batteryMode == o, onSelected: (_) => notifier.setBatteryMode(o), selectedColor: colors.primaryContainer, side: BorderSide.none, shape: RoundedRectangleBorder(borderRadius: NanoShapes.full))))).toList())),
        const SizedBox(height: NanoSpacing.xxxl),
      ],
    );
  }

  Widget _h(String t, IconData i, NanoColors c) => Padding(padding: const EdgeInsets.only(bottom: NanoSpacing.sm), child: Row(children: [Icon(i, size: 16, color: c.primary), const SizedBox(width: NanoSpacing.sm), Text(t.toUpperCase(), style: TextStyle(fontSize: 11, fontWeight: FontWeight.w700, color: c.onSurfaceVariant, letterSpacing: 0.8))]));
  Widget _sw(String t, String s, bool v, Function(bool) cb, NanoColors c) => Padding(padding: const EdgeInsets.symmetric(vertical: NanoSpacing.sm), child: Row(children: [Expanded(child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [Text(t, style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500, color: c.onSurface)), Text(s, style: TextStyle(fontSize: 11, color: c.onSurfaceVariant))])), Switch(value: v, onChanged: cb, activeTrackColor: c.primary.withValues(alpha: 0.4))]));
  Widget _sl(String t, double v, double min, double max, String u, Function(double) cb, NanoColors c) => Padding(padding: const EdgeInsets.symmetric(vertical: NanoSpacing.sm), child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [Row(mainAxisAlignment: MainAxisAlignment.spaceBetween, children: [Text(t, style: TextStyle(fontSize: 14, fontWeight: FontWeight.w500, color: c.onSurface)), Text('${v.toStringAsFixed(1)} $u', style: TextStyle(fontSize: 14, fontWeight: FontWeight.w600, color: c.primary))]), Slider(value: v, min: min, max: max, onChanged: cb, activeColor: c.primary, inactiveColor: c.outlineVariant.withValues(alpha: 0.5))]));
}
